test/jdk/java/net/httpclient/websocket/security/WSURLPermissionTest.java

```diff
@@ -359,190 +339,7 @@ public void testWithNoSecurityManager(PrivilegedExceptionAction<?> action,
-              new URLPermission[] { new URLPermission(wsURI.resolve("/aDiffPath/").toString()) },
-              "57" },
-
-            { (PrivilegedExceptionAction<?>)() -> {
-                URI uriWithPath = wsURI.resolve("/path/x");
-                 noProxyClient.newWebSocketBuilder()
-                              .buildAsync(uriWithPath, noOpListener).get().abort();
-                 return null; },                                    // more specific path
-              new URLPermission[] { new URLPermission(wsURI.resolve("/path/x/y").toString()) },
-              "58" },
-
-            // client with a HTTP/HTTPS proxy
-            { (PrivilegedExceptionAction<?>)() -> {
-                 assert proxyAddress != null;
-                 ProxySelector ps = ProxySelector.of(proxyAddress);
-                 HttpClient client = HttpClient.newBuilder().proxy(ps).build();
-                 client.newWebSocketBuilder()
-                       .buildAsync(wsURI, noOpListener).get().abort();
-                 return null; },                                    // missing proxy perm
-              new URLPermission[] { new URLPermission(wsURI.toString()) },
-              "100" },
-
-            // client with a HTTP/HTTPS proxy
-            { (PrivilegedExceptionAction<?>)() -> {
-                 assert proxyAddress != null;
-                 ProxySelector ps = ProxySelector.of(proxyAddress);
-                 HttpClient client = HttpClient.newBuilder().proxy(ps).build();
-                 client.newWebSocketBuilder()
-                       .buildAsync(wsURI, noOpListener).get().abort();
-                 return null; },
-              new URLPermission[] {
-                    new URLPermission(wsURI.toString()),            // missing proxy CONNECT
-                    new URLPermission("socket://*", "GET") },
-              "101" },
-        };
-    }
-
-    @Test(dataProvider = "failingScenarios")
-    public void testWithoutEnoughPermissions(PrivilegedExceptionAction<?> action,
-                                             URLPermission[] perms,
-                                             String dataProviderId)
-        throws Exception
-    {
-        // Run without Enough permissions, i.e. expect SecurityException
-        assert System.getSecurityManager() != null;
-        AccessControlContext notEnoughPermsACC = withPermissions(perms);
-        try {
-            AccessController.doPrivileged(action, notEnoughPermsACC);
-            fail("EXPECTED SecurityException");
-        } catch (PrivilegedActionException expected) {
-            Throwable t = expected.getCause();
-            if (t instanceof ExecutionException)
-                t = t.getCause();
-
-            if (t instanceof SecurityException)
-                System.out.println("Caught expected SE:" + expected);
-            else
-                fail("Expected SecurityException, but got: " + t);
-        }
+        action.run();
```

Comment:
testWithNoSecurityManager was previously a sanity check, the test was focused on permission check. Is the test still useful to keep, maybe it would be renamed or the test method renamed?

Reply:
Good point. Similarly, the URLPermission[] parameter is now always unused, so maybe I should get rid of that too.

Reply:
Fixed in https://github.com/openjdk/jdk/pull/21498/commits/82bb0d8207334d6072277e596fb16228f397fb77 and https://github.com/openjdk/jdk/pull/21498/commits/34439751f1b26e6ff1705e35f269e260401233af